### ISSUE TITLE
fine-grained authorization with JWT token

### ISSUE THREAD

**Issue by stevehu** [networknt/light-graphql-4j#27]
Unlike RESTful API, we have scope for each endpoint so that the authorization can be down to the endpoint. For GraphQL, there is only one endpoint with only two or three different type of requests. Need to find a way to protect individual query or mutation.

**Comment by maarek** [comment]
Can you expose the request to the execution context so that DataFetchers gain access to the JWT token during a request and the authentication is deferred to the business layer?

**Comment by stevehu** [comment]
Yes. That is a very good idea. In the light-rest-4j we pass the payload of JWT in an exchange attachment for fine-grained authorization in the business context. Let me explore it a little bit and propose something for further discussion. Thanks.
